### PR TITLE
Machina sate machine

### PR DESCRIPTION
**Pull request by qstearns**
Updated the state machine. Design changes and further abstraction to overall Turn class are needed. Current design is fundamentally flawed in that the FSM is a member of the MonopolyTurn class, which is a descendent of the Turn class. It is possible that the design is reasonable and there better naming consistency would repair the issue, but further discussion is needed to be confident

### CODE REVIEW

File: src/game.js [hunk 1]
```diff
@@ -30,9 +30,9 @@ Game.prototype.isDoubles = function(dice) {
   return dice[0] === dice[1];
 };
 
-Game.prototype.setState = function(newState) {
-  this.state = newState;
-};
+// Game.prototype.setState = function(newState) {
```
Comment (konnorbeard) [inline src/game.js]: commented code

File: src/monopoly.js [hunk 2]
```diff
@@ -23,16 +23,20 @@ var players = [john, steve, sam, mike, jimmy];
...
+
+turn.turnMap.on("transition", function (data){
+    console.log("we just transitioned from " + data.fromState + " to " + data.toState);
+});
 
 view.drawBoard();
 
 //start running game
-turn.runStateMachine(false, monopoly);
+// turn.runStateMachine(false, monopoly);
```
Comment (konnorbeard) [inline src/monopoly.js]: try to delete commented code/logging stuff unless we need it for later

**Comment by konnorbeard** [comment]
@qstearns this looks great, only comments i have are to either remove the commented code/logging or explain why it's there. Otherwise great work

File: index.html [hunk 3]
```diff
@@ -2,7 +2,6 @@
 <html>
 <head>
   <meta charset="UTF-8">
-  <script type="text/javascript" src="lib/pixi/pixi.min.js"></script>
```
Comment (andrewjmeier) [inline index.html]: What happened to Pixi?

File: src/monopoly/view/monopoly_view.js [hunk 4]
```diff
@@ -1,3 +1,5 @@
+var PIXI = require("../../../lib/pixi/pixi.js")
```
Comment (andrewjmeier) [inline src/monopoly/view/monopoly_view.js]: Oh got it here!

**Comment by andrewjmeier** [comment]
Cool lets try and get this merged in and then we can do the little updates to trading that we need and get that merged in too.

**Pull request merged**